File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/AuthorizationSample/src/com/hp/workpath/sample/authorization/task/GetSignInMethodsTask.java
// Copyright 2025 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.authorization.task;

import android.content.Context;
import android.text.TextUtils;

import com.hp.workpath.api.Result;
import com.hp.workpath.api.authorization.AuthorizationService;
import com.hp.workpath.api.authorization.SignInMethod;
import com.hp.workpath.sample.authorization.R;
import com.hp.workpath.sample.authorization.exception.ResultException;

import java.lang.ref.WeakReference;
import java.util.ArrayList;
import java.util.concurrent.Callable;

public class GetSignInMethodsTask implements Callable<ArrayList<SignInMethod>> {

    WeakReference<Context> context;
    String languageCode;

    public GetSignInMethodsTask(Context context, String languageCode) {
        this.context = new WeakReference<>(context);
        this.languageCode = languageCode;
    }

    @Override
    public ArrayList<SignInMethod> call() throws Exception {
        Result result = new Result();
        if (TextUtils.isEmpty(languageCode)) {
            languageCode = context.get().getString(R.string.en_us);
        }
        ArrayList<SignInMethod> signInMethods = AuthorizationService.getSignInMethod(context.get(), result, languageCode);
        if (result.getCode() != Result.RESULT_OK) {
            throw new ResultException(result);
        }
        return signInMethods;
    }
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/AuthorizationSample/src/com/hp/workpath/sample/authorization/task/GetSignInMethodsTask.kt
// Copyright 2025 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.authorization.task

import android.content.Context
import com.hp.workpath.api.Result
import com.hp.workpath.api.authorization.AuthorizationService
import com.hp.workpath.api.authorization.SignInMethod
import com.hp.workpath.sample.authorization.R
import com.hp.workpath.sample.authorization.exception.ResultException
import java.lang.ref.WeakReference
import java.util.concurrent.Callable

class GetSignInMethodsTask(context: Context, private var languageCode: String?) :
    Callable<ArrayList<SignInMethod>?> {

    private val contextRef: WeakReference<Context> = WeakReference(context)

    @Throws(Exception::class)
    override fun call(): ArrayList<SignInMethod>? {
        val result = Result()
        val context = contextRef.get()
        if (languageCode.isNullOrEmpty()) {
            languageCode = context?.getString(R.string.en_us) ?: ""
        }
        val signInMethods = context?.let { ctx ->
            AuthorizationService.getSignInMethod(ctx, result, languageCode ?: "")
        }
        if (result.code != Result.RESULT_OK) {
            throw ResultException(result)
        }
        return signInMethods
    }
}